src/WaveTrack.cpp

```diff
@@ -1408,7 +1412,11 @@ void WaveTrack::Paste(double t0, const Track *src)
             newClip->Resample(mRate);
             newClip->Offset(t0);
             newClip->MarkChanged();
-            newClip->SetName(MakeClipCopyName(clip->GetName()));
+            if (pastingFromClipboard)
+               newClip->SetName(MakeClipCopyName(clip->GetName()));
+            else if (pastingFromTempTrack)
+               newClip->SetName(MakeNewClipName());
```

Comment:
Hm, Silence is not the only generator which produces clips with invalid names, other generators are affected too (but they create clips a bit differently)

Reply:
How do you generate clips with bad names with other generators?

Reply:
I see, MakeClipCopyName generates names containing "copy" when it should not.

You want that only when pasting clips that really come from the clipboard -- not for clips that are also pasted into a track because effects are implemented that way.

So:  this does go beyond fixing the issue as stated.  You need to distinguish clips from the clipboard so that only those get "copy" in their names.

I would still rather let clips carry some name-is-valid flag in them.

Reply:
Then I think it can be simplified: if the track has the owner then use the 'copy' name pattern, if not - use the 'new' name pattern.